app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb

```diff
@@ -248,6 +248,27 @@ def vm_reconfigure(vm, options = {})
       _log.info("#{log_header} Completed.")
     end
 
+    def advertised_images
+      ext_management_system.with_provider_connection(VERSION_HASH) do |ems_service|
+        query = { :search => "status=#{OvirtSDK4::DataCenterStatus::UP}" }
+        data_centers = ems_service.system_service.data_centers_service.list(:query => query)
+        iso_sd = data_centers.each do |dc|
+          active_iso_sd = ems_service.follow_link(dc.storage_domains).detect do |sd|
+            sd.type == OvirtSDK4::StorageDomainType::ISO && sd.status == OvirtSDK4::StorageDomainStatus::ACTIVE
+          end
+          break active_iso_sd if active_iso_sd
+        end
+
+        sd_service = ems_service.system_service.storage_domains_service.storage_domain_service(iso_sd.id)
```

Comment:
we should take into account the possibility of iso_sd is nil, in case of no DC up, or in case of no Storage Domain was defined on RHV.

We should return from this method with an empty list, and should log that "no active ISO data store was found" message.